src/applications/disability-benefits/2346/schemas/address-schema.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * The intent for this module is to provide a flexible, reusable address schema and widget that can be used in any form throughout VA.gov.
+ * The address uiSchema should be flexible enough to handle these cases:
+ * 1. Top level address property (schema.properties.address)
+ * 2. Nested address property (schema.properties.someProperty.properties.address)
+ * 3. Array items.
+ *
+ * Fields that may depend on external variables to make the form required:
+ * 1. Country - could depend on things like: yes/no field, checkbox in a different form chapter, etc.
+ * 2. Address Line 1 - same as country
+ * 3. City - same as country
+ *
+ * Fields that are required based on internal field variables:
+ * 1. State - required if the country is the United States OR US Military Address
+ * 2. Zipcode - required if the country is the United States OR US Military Address
+ * 3. International Postal Code - required if the country is NOT the United States OR US Military address
+ *
+ * Fields that are optional:
+ * 1. State/Province/Region - shows up if the country is NOT the US, but NOT required.
+ */
+
+import AdditionalInfo from '@department-of-veterans-affairs/formation-react/AdditionalInfo';
+import { countries } from 'platform/forms-system/src/js/utilities/address';
+import ADDRESS_DATA from 'platform/forms/address/data';
+import cloneDeep from 'platform/utilities/data/cloneDeep';
+import get from 'platform/utilities/data/get';
+import React from 'react';
+import { militaryCities, states50AndDC } from '../constants';
+
+/**
+ * CONSTANTS:
+ * 1. MILITARY_STATES - object of military state codes and names.
+ * 2. USA - used to just reference the United States
+ * 3. MilitaryBaseInfo - expandable text to expound on military base addresses.
+ * 4. addressSchema - data model for address schema.
+ */
+
+const MILITARY_STATES = Object.entries(ADDRESS_DATA.states).reduce(
+  (militaryStates, [stateCode, stateName]) => {
+    if (ADDRESS_DATA.militaryStates.includes(stateCode)) {
+      return {
+        ...militaryStates,
+        [stateCode]: stateName,
+      };
+    }
+    return militaryStates;
+  },
+  {},
+);
+
+const USA = {
+  value: 'USA',
+  name: 'United States',
+};
+
+const MilitaryBaseInfo = () => (
+  <div className="vads-u-padding-x--2p5">
```

Comment:
We could omit this wrapping div and add a className to `AdditionalInfo` below if you want! 🙂